### PR TITLE
fix: make all options optionally

### PR DESCRIPTION
**Pull request by HashimotoYT**
e2e testing is the best testing 🙈

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 declare namespace hubdown {
   interface Options {
-    readonly runBefore: Array<any>;
-    readonly frontmatter: boolean;
-    readonly cache: any;
+    readonly runBefore?: Array<any>;
```
Comment (ChristianMurphy) [inline index.d.ts]: If you're interested, [`unified`](https://github.com/unifiedjs/unified) (and by extension [`remark`](https://github.com/remarkjs/remark)) has a [`Plugin` type](https://github.com/unifiedjs/unified/blob/12e0adc5b5240ebf1ec378a22a7b6e2f64a266d7/types/index.d.ts#L205-L217) that could be used here.

File: index.d.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+    readonly frontmatter?: boolean;
+    readonly cache?: any;
```
Comment (ChristianMurphy) [inline index.d.ts]: Not available yet, but https://github.com/Level/level/issues/120 may someday offer a type the could be used here.

File: index.d.ts (hunk 1 above)
Comment (ChristianMurphy) [reply index.d.ts]: In this case https://github.com/unifiedjs/unified/blob/12e0adc5b5240ebf1ec378a22a7b6e2f64a266d7/types/index.d.ts#L273-L278 may be more accurate

```ts
import {PluggableList} = require('unified');
import {RemarkOptions} = require('remark');

// ...
   readonly runBefore?: PluggableList<Partial<RemarkOptions>>;
```

**Comment by HashimotoYT** [comment]
@zeke It's ready, I'm merging this when being ready.
@CharlieHess thanks for typings suggestions, I'm happy to review this in the other PR.

**Pull request merged**